hibernate-cache/src/test/java/org/infinispan/test/hibernate/cache/functional/AbstractFunctionalTest.java

```diff
@@ -239,4 +253,114 @@ protected CountDownLatch expectPutWithValue(AdvancedCache cache, Predicate<Objec
 			return new CountDownLatch(0);
 		}
 	}
+
+   protected CountDownLatch expectAfterEndInvalidation(AdvancedCache cache, int numInvalidates) {
+      CountDownLatch latch = new CountDownLatch(numInvalidates);
+      wrapInboundInvocationHandler(cache, handler -> new ExpectingInboundInvocationHandler(handler, latch));
```

Comment:
> I suggest adding a List<Runnable> cleanup collection as a field and executing tasks from that from @AfterMethod.

I disagree with that because it's not symmetric.

I do agree with the need to remove the handler, but it should be symmetric. If the handler is added in the test method, it should be removed in the test method itself.